File: medium/3719-longest-balanced-subarray-i/solution.java
import java.util.HashSet;
import java.util.Set;

/**
 * 3719. Longest Balanced Subarray I
 * Time: O(n²)
 * Space: O(n)
 */
class Solution {
    public int longestBalancedSubarray(int[] nums) {
        int n = nums.length;
        int ans = 0;

        for (int i = 0; i < n; i++) {
            Set<Integer> even = new HashSet<>();
            Set<Integer> odd = new HashSet<>();

            for (int j = i; j < n; j++) {
                if (nums[j] % 2 == 0) {
                    even.add(nums[j]);
                } else {
                    odd.add(nums[j]);
                }

                if (even.size() == odd.size()) {
                    ans = Math.max(ans, j - i + 1);
                }
            }
        }

        return ans;
    }
}

File: medium/3719-longest-balanced-subarray-i/solution.py
from typing import List


class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        """
        3719. Longest Balanced Subarray I
        Time: O(n²)
        Space: O(n)
        """
        n = len(nums)
        ans = 0

        for i in range(n):
            even = set()
            odd = set()

            for j in range(i, n):
                if nums[j] % 2 == 0:
                    even.add(nums[j])
                else:
                    odd.add(nums[j])

                if len(even) == len(odd):
                    ans = max(ans, j - i + 1)

        return ans
